pulseguardian/web.py

```diff
@@ -277,40 +279,65 @@ def bindings_listing(queue_name):
 
 # Authentication related
 
-@app.route('/auth/login', methods=['POST'])
-def auth_handler():
-    # The request has to have an assertion for us to verify
-    if 'assertion' not in request.form:
-        return jsonify(ok=False, message="Assertion parameter missing")
+@app.route('/auth/callback')
+def callback_handling():
+    """
+    Callback from Auth0
 
-    # Send the assertion to Mozilla's verifier service.
-    data = dict(assertion=request.form['assertion'],
-                audience=config.persona_audience)
-    resp = requests.post(config.persona_verifier, data=data, verify=True)
+    Auth0 will call back to this endpoint once it has acquired a user in some
+    capacity (from Google, or their email).  Here we verify what we got back
+    it consistent with Auth0, then match what we got back
+    against our internal DB and either find the user, or create it.
+    """
 
-    # Did the verifier respond?
+    # validate what we got in the request against the Auth0 backend
+    code = request.args.get('code')
+    json_header = {'content-type': 'application/json'}
+    token_url = "https://{domain}/oauth/token".format(domain=config.auth0_domain)
+    token_payload = {
+      'client_id': config.auth0_client_id,
+      'client_secret': config.auth0_client_secret,
+      'redirect_uri': config.auth0_callback_url,
+      'code': code,
+      'grant_type': 'authorization_code'
+    }
+    token_info = requests.post(token_url, data=json.dumps(token_payload), headers = json_header).json()
+    user_url = "https://{domain}/userinfo?access_token={access_token}"  \
```

Comment:
Rather than using a backslash, it would be better to end this line on the opening parenthesis of `format(` (and then put the closing parenthesis on its own line).